forex_python/converter.py

```diff
@@ -79,35 +87,61 @@ def get_rate(self, base_cur, dest_cur, date_obj=None):
         raise RatesNotAvailableError("Currency Rates Source Not Ready")
 
     def convert(self, base_cur, dest_cur, amount, date_obj=None):
+        use_decimal = False
         if isinstance(amount, Decimal):
             use_decimal = True
-        else:
-            use_decimal = self._force_decimal
+        elif self._force_decimal:
+            raise DecimalFloatMismatchError("Decimal is forced. Amount must be Decimal")
 
         if base_cur == dest_cur:  # Return same amount if both base_cur, dest_cur are same
-            if use_decimal:
-                return Decimal(amount)
-            return float(amount)
+            return amount
+        
+        rate = self.get_rate(base_cur, dest_cur, date_obj=date_obj, use_decimal=use_decimal)
 
+        return rate * amount
+
+class CurrencyRatesForexAPI(CurrencyRatesBase):
+    def _source_url(self):
+        return "https://theforexapi.com/api/"
+
+class CurrencyRatesExchangeRateHost(CurrencyRatesBase):
+    def _source_url(self):
+        return "https://api.exchangerate.host/"
+    
+    def get_rates(self, base_cur, date_obj=None):
         date_str = self._get_date_string(date_obj)
-        payload = {'base': base_cur, 'symbols': dest_cur, 'rtype': 'fpy'}
+        payload = {'base': base_cur, 'rtype': 'fpy'}
         source_url = self._source_url() + date_str
-        response = requests.get(source_url, params=payload)
+        response = self._request(source_url, params=payload)
         if response.status_code == 200:
-            rate = self._get_decoded_rate(
-                response, dest_cur, use_decimal=use_decimal, date_str=date_str)
-            if not rate:
-                raise RatesNotAvailableError("Currency {0} => {1} rate not available for Date {2}.".format(
-                    source_url, dest_cur, date_str))
-            try:
-                converted_amount = rate * amount
-                return converted_amount
-            except TypeError:
-                raise DecimalFloatMismatchError(
-                    "convert requires amount parameter is of type Decimal when force_decimal=True")
+            rates = self._decode_rates(response,date_str=date_str, base_cur=base_cur)
+            return rates
```

Comment:
Pulled this implementation from this PR: https://github.com/MicroPyramid/forex-python/pull/127